azurelinuxagent/ga/exthandlers.py

```diff
@@ -362,9 +360,37 @@ def handle_ext_handlers(self, etag=None):
 
         self.ext_handlers.extHandlers.sort(key=operator.methodcaller('sort_key'))
         for ext_handler in self.ext_handlers.extHandlers:
-            # TODO: handle install in sequence, enable in parallel
+            self.wait_on_ext_handler_dependencies(ext_handler)
```

Comment:
At least to me, this does not seem like an intuitive way of handling dependencies. In fact, they really are not dependencies, just a particular sequence. Why the need to resolve dependencies and build another chain, when the parent loop has all the info it needs? If the order of `self.ext_handlers.extHandlers` mapped to the dependency order, would that not do the same thing? 

It seems like the intent here is to have each link in the chain know something about their dependency (or really, their pre-cursor) and that seems to be just so we can query the status? We can do that from this loop more simply, imo. Let me know if I have misunderstood the intention here though.

Reply:
Let's say we have 6 extension handlers (1 extension per handler): A, B, C, D, E, F.
D depends on A, E depends on B, F depends on C. The order is still: A, B, C, D, E, F
When A finishes, B can run right away. When B finishes, C can run right away. They do not depend on anything. D has to ensure that A was successful before it can run. If A and B were successful, but C was not, then D and E can run but F cannot. It is, of course, possible that A will fail and prevent all of D, E, F from running, but that is still a better outcome than A failing and B, C, D, E, F being prevented from running.